### ISSUE TITLE
Compilation error: illegal use of incomplete type!

### ISSUE THREAD

**Issue by sebleblanc** [kaoskorobase/hsndfile#18]
Compilation fails with the following error:

```
c2hs: Errors during expansion of binding hooks:

/usr/include/sdnfile.h:359: (column 1) [ERROR]  >>> Illegal use of incomplete type!
 Expected a fully defined structure or union tag; instead found incomplete type.
```

Meanwhile, browsing to the definition and replacing it with the following seems to correct the issue:

```
typedef struct SF_INFO
{ ...
} SF_INFO;
```

**Comment by kaoskorobase** [comment]
Which version of c2hs are you using? This issue seems to have appeared with c2hs version 0.26 and the workaround was to use version 0.25.2, see #14 for the discussion.

**Comment by kaoskorobase** [comment]
Apparently this has been fixed in c2hs 0.27, see https://github.com/haskell/c2hs/issues/152

**Comment by kaoskorobase** [comment]
Closing for now, feel free to reopen if you still encounter the same problem with c2hs >= 0.27.
